docs/workflow.md

```diff
@@ -22,18 +22,69 @@ workflow {
 }
 ```
 
-### Parameters
+(workflow-params-def)=
 
-Parameters can be defined in the script with a default value that can be overridden from the CLI, params file, or config file. Parameters should only be used by the entry workflow:
+## Parameters
+
+Parameters can be declared in a Nextflow script with the `params` block or with *legacy* parameter declarations.
+
+### Params block
+
+:::{versionadded} 25.04.0
+:::
+
+:::{note}
+This feature requires the `nextflow.preview.params` feature flag to be enabled.
+:::
+
+A script can declare parameters using the `params` block:
+
+```nextflow
+params {
+    /**
+     * Path to input data.
+     */
+    input
+
+    /**
+     * Whether to save intermediate files.
+     */
+    save_intermeds = false
+}
+```
+
+Parameters can be used in the entry workflow:
+
+```nextflow
+workflow {
+    if( params.input )
+        analyze(params.input, params.save_intermeds)
+    else
+        analyze(fake_input(), params.save_intermeds)
+}
+```
+
+:::{note}
+Named workflows should not use params. They should receive their inputs explicitly through the `take:` section.
```

Comment:
Not sure about this. I think it should be maintained the use of parameters both in workflows and processes as should as they are immutable

Reply:
To be clear, parameters can still be used everywhere. This is just a recommendation since params cannot be validated outside the entry workflow. I can document this more explicitly if you want

Reply:
I have reframed this note as a best practice recommendation, since it is still allowed